ariadne/cardano/test/backend/Test/Spec/Fixture.hs

```diff
@@ -39,13 +43,27 @@ genSpendingPassword :: PropertyM IO PassPhrase
 genSpendingPassword =
     pick (frequency [(20, pure emptyPassphrase), (80, arbitrary)])
 
+genSpendingPasswords :: Int -> PropertyM IO [PassPhrase]
+genSpendingPasswords n = pick $ vector n
+
 withLayer :: MonadIO m
           => ProtocolMagic
           -> (PassiveWalletLayer m -> Kernel.PassiveWallet -> IO a)
           -> PropertyM IO a
-withLayer pm cc = do
+withLayer pm cc =
     liftIO $ Keystore.bracketTestKeystore $ \keystore -> do
-        bracketKernelPassiveWallet pm devNull keystore $ \layer wallet -> do
+        bracketKernelPassiveWallet pm devNull keystore Memory $ \layer wallet -> do
+            cc layer wallet
+
+withLayerLocalStorage
+  :: MonadIO m
+  => ProtocolMagic
+  -> FilePath
+  -> (PassiveWalletLayer m -> Kernel.PassiveWallet -> IO a)
+  -> m a
+withLayerLocalStorage pm pathToDB cc =
+    liftIO $ Keystore.bracketTestKeystore $ \keystore -> do
+        bracketKernelPassiveWallet pm devNull keystore (Filesystem pathToDB) $ \layer wallet -> do
```

Comment:
I don't like this copy-paste. Why not make a helper function with type `MonadIO m => ProtocolMagic -> FileOrMemoryDB -> (PassiveWalletLayer m -> Kernel.PassiveWallet -> IO a) -> m a`? And then `withLayerLocalStorage` and `withLayer` can use it (feel free to rename smth).

Reply:
Ok.